File: Questions/UnixComms.java
import java.util.*;

public class UnixComms {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();  // consume the newline after the integer input
        String[] cmd = new String[n];
        for (int i = 0; i < n; i++) {
            cmd[i] = sc.nextLine();
        }

        int[] ans = func(cmd);
        for (int var : ans) {
            System.out.print(var + " ");
        }
    }

    public static int[] func(String[] cmd) {
        int l = cmd.length;
        int[] res = new int[3]; // Array to hold counts for "cp", "mv", "ls"

        for (int i = 0; i < l; i++) {
            switch (cmd[i]) {
                case "ls":
                    res[0]++;
                    break;
                case "cp":
                    res[1]++;
                    break;
                case "mv":
                    res[2]++;
                    break;
                default:
                    boolean looper = false;
                    int index = Integer.parseInt(cmd[i].substring(1)) - 1;
                    while (!looper) {
                        
                    System.out.println("______________CHECK_______________");
                    System.out.println(cmd[i]);
                    System.out.println(index);
                    System.out.println(cmd[index]); // Get the index from the command
                    System.out.println("______________END_CHECK_______________");
                    
                    String s = cmd[index];
                        if (s.charAt(0) == 'l') {
                            res[0]++;
                            looper = true;
                        } else if (s.charAt(0) == 'c') {
                            res[1]++;
                            looper = true;
                        } else if (s.charAt(0) == 'm') {
                            res[2]++;
                            looper = true;
                        } else {
                            index = Integer.parseInt(s.substring(1)) -1;

                        }
                    }
                    break;
            }
        }
        return res;
    }
}

File: Questions/Python/UnixComms.py
arr = list(map(str, input().split()))

command_counts = {}
result = [0, 0, 0]
for i in arr:
    if i == "cp":
        command_counts["cp"] = command_counts.get("cp", 0) + 1
        result[0] += 1
    elif i == "mv":
        command_counts["mv"] = command_counts.get("mv", 0) + 1
        result[1] += 1
    elif i == "ls":
        command_counts["ls"] = command_counts.get("ls", 0) + 1
        result[2] += 1
    else:
        index = int(i[1])
        while True:
            if arr[index] == "c":
                command_counts["cp"] = command_counts.get("cp", 0) + 1
                result[0] += 1
                break
            elif arr[index] == "m":
                command_counts["mv"] = command_counts.get("mv", 0) + 1
                result[1] += 1
                break
            elif arr[index] == "l":
                command_counts["ls"] = command_counts.get("ls", 0) + 1
                result[2] += 1
                break
            else:
                index = int(arr[index][1])

print(command_counts)
print(result)


# arr = list(map(str,input().split()))

# map = {}
# res = [0,0,0]
# for i in arr:
#     if i == "cp":
#         map["cp"] = map.get("cp", 0) + 1
#         res[0] +=1
#     elif i == "mv":
#         map["mv"] = map.get("mv", 0) + 1
#         res[1] +=1
#     elif i == "ls":
#         map["ls"] = map.get("ls", 0) + 1
#         res[2] +=1
#     else:
#         index = i[1]
#         looper = True
#         while looper:
#             if arr[index] == "c":
#                 map["cp"] = map.get("cp", 0) + 1
#                 res[0] +=1
#                 looper = False
#                 break
#             elif arr[index] == "m":
#                 map["mv"] = map.get("mv", 0) + 1
#                 res[1] +=1
#                 looper = False
#                 break
#             elif arr[index] == "l":
#                 map["ls"] = map.get("ls", 0) + 1
#                 res[2] +=1
#                 looper = False
#                 break
#             else:
#                 index = arr[index]
#                 index = index [1]
#                 looper = True
# print(map)
# print(res)
# #